stormevents/nhc/track.py

```diff
@@ -1072,17 +1072,25 @@ def __compute_velocity(data: DataFrame) -> DataFrame:
         for advisory in pandas.unique(data["advisory"]):
             advisory_data = data.loc[data["advisory"] == advisory]
 
-            indices = numpy.array(
-                [
-                    numpy.where(advisory_data["datetime"] == unique_datetime)[0][0]
-                    for unique_datetime in pandas.unique(advisory_data["datetime"])
-                ]
-            )
+            indices = advisory_data.index
             shifted_indices = numpy.roll(indices, 1)
-            shifted_indices[0] = 0
-
-            indices = advisory_data.index[indices]
-            shifted_indices = advisory_data.index[shifted_indices]
+            shifted_indices[0] = indices[0]
+
+            # check for negative time shifts which indicate new forecasts
+            # and update this with the last previously available time
+            for counter, ind in enumerate(zip(indices, shifted_indices)):
+                this_time = advisory_data.loc[ind[0], "datetime"]
+                shift_time = advisory_data.loc[ind[1], "datetime"]
+                if shift_time > this_time:
+                    # update shift index
+                    if (advisory_data["datetime"] < this_time).sum() == 0:
+                        shifted_indices[counter] = advisory_data["datetime"][
+                            advisory_data["datetime"] > this_time
+                        ].index[0]
```

Comment:
@WPringle can you please explain what this does? It says that if there's no time in the dataframe smaller than the current row we're processing, then the shifted time ("previous time") should be the first time that is larger than the current time. Why?! This gives us opposite bearing, right?

Reply:
That's a good point about the bearing! This could be a bug that needs fixing.

Reply:
It seems all the tests are passing. The coverage test runs into a timeout issue. Do you want me to go ahead with this merge and then later explore the bearing issue in a separate ticket or do you want me to wait for you to test the potential bearing issue?

Reply:
Thanks for this! Please wait until I double check everything. I will let you know once happy.

Reply:
OK for the bearing, in this case we can use the `forward_azimuth` instead of `inverse_azimuth`. The change has been added in latest commit. Note this is overall a rare case.